### PR TITLE
tracing: Make collection of opentelemetry traces optional

### PR DESCRIPTION
**Pull request by ashishb-solo**
This commit marks the `grpc_service` of the opentelemtry configuration as optional and if the resulting field is empty, the plugin will abstain from sending the trace data to any collection service.

This means that the opentelemetry plugin will still generate and propagate trace headers, but they will no longer be sent to the collector.

Signed-off-by: Ashish Banerjee <ashish.banerjee@solo.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/tracers/opentelemetry/config_test.cc [hunk 1]
```diff
@@ -41,6 +41,27 @@ TEST(OpenTelemetryTracerConfigTest, OpenTelemetryHttpTracer) {
...
+      typed_config:
+        "@type": type.googleapis.com/envoy.config.trace.v3.OpenTelemetryConfig
+  )EOF";
+  envoy::config::trace::v3::Tracing configuration;
+  TestUtility::loadFromYaml(yaml_string, configuration);
+
+  auto message = Config::Utility::translateToFactoryConfig(
+      configuration.http(), ProtobufMessage::getStrictValidationVisitor(), factory);
+  auto opentelemetry_tracer = factory.createTracerDriver(*message, context);
+  std::cout << typeid(opentelemetry_tracer).name() << std::endl;
```
Comment (lizan) [inline test/extensions/tracers/opentelemetry/config_test.cc:61]: Remove?

**Review by lizan — COMMENTED** [review]
LGTM, just one nit

File: test/extensions/tracers/opentelemetry/config_test.cc (hunk 1 above)
Comment (ashishb-solo) [reply test/extensions/tracers/opentelemetry/config_test.cc]: https://github.com/envoyproxy/envoy/pull/24792/commits/6657bb606bc042c08494ed47fe7f2cb9472ed40a

thanks for the review!

**Comment by ashishb-solo** [comment]
Hi @lizan - not sure if you saw, but I updated the pull request as you suggested. Would appreciate if you could take one more look. Thanks!

**Review by lizan — DISMISSED** [review]

**Comment by lizan** [comment]
@ashishb-solo sorry for the delay, can you merge main branch to resolve the conflicts?

**Comment by ashishb-solo** [comment]
@lizan I just merged with the main branch, could you check it over again! Thanks

**Review by lizan — APPROVED** [review]

**Pull request merged**